pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -157,14 +213,65 @@ type ContolPlaneJoinInput struct {
 	CACertHash     string
 	CACert         string
 	CAKey          string
+	EtcdCert       string
+	EtcdKey        string
+	FrontProxyCert string
+	FrontProxyKey  string
+	SaCert         string
+	SaKey          string
 	BootstrapToken string
 	ELBAddress     string
-	KubeConfig     string
+}
+
+func (cpi *ControlPlaneInput) isValid() (bool, string) {
+	// TODO: ashish-amarnath verify if ca cert and key is mandatory for kubeadm init
```

Comment:
kubeadm init will succeed without a ca certificate, but in cluster-api-provider-aws we provide the CA cert so we can store it on the ClusterSpec and share it across control planes.